### PR TITLE
Dont use ASAN for benchmark builds

### PR DESCRIPTION
**Pull request by jonathanmetzman**
We aren't detecting crashes yet so it isn't very useful.

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
@inferno-chromium Could you take this over please?

File: fuzzers/afl/fuzzer.py [hunk 1]
```diff
@@ -24,10 +24,8 @@
 
 def prepare_build_environment():
     """Set environment variables used to build AFL-based fuzzers."""
-    cflags = [
-        '-O2', '-fno-omit-frame-pointer', '-gline-tables-only',
-        '-fsanitize=address', '-fsanitize-coverage=trace-pc-guard'
-    ]
+    utils.set_no_sanitizer_compilation_flags()
+    cflags = ['-O3' '-fsanitize-coverage=trace-pc-guard']
```
Comment (inferno-chromium) [inline fuzzers/afl/fuzzer.py]: Why some has O3 vs others have O2, we should standardize on one.

**Comment by robertswiecki** [comment]
Should this also change honggfuzz's compilation flags. In https://github.com/google/fuzzbench/pull/129 the build seem to be generated correctly w/o asan.

**Comment by inferno-chromium** [comment]
Yes looks like we missed Honggfuzz change here. i recounted the things, fastcgs is not required, rest all fuzzers are here except Honggfuzz.

**Comment by jonathanmetzman** [comment]
@morehouse @kcc Does not using ASAN hurt libFuzzer? I vaguely remember something about ASAN giving libFuzzer the ability to instrument more of the runtime when ASAN is used. WDYT about not using ASAN for libFuzzer if we aren't going to use it for any other fuzzer?

**Comment by morehouse** [comment]
> @morehouse @kcc Does not using ASAN hurt libFuzzer? I vaguely remember something about ASAN giving libFuzzer the ability to instrument more of the runtime when ASAN is used. WDYT about not using ASAN for libFuzzer if we aren't going to use it for any other fuzzer?

I'm pretty sure some important libFuzzer heuristics rely on ASan hooks for memcmp/strcmp and friends.  https://github.com/llvm-mirror/compiler-rt/blob/master/lib/fuzzer/FuzzerTracePC.cpp#L585

**Comment by jonathanmetzman** [comment]
> > @morehouse @kcc Does not using ASAN hurt libFuzzer? I vaguely remember something about ASAN giving libFuzzer the ability to instrument more of the runtime when ASAN is used. WDYT about not using ASAN for libFuzzer if we aren't going to use it for any other fuzzer?
> 
> I'm pretty sure some important libFuzzer heuristics rely on ASan hooks for memcmp/strcmp and friends. https://github.com/llvm-mirror/compiler-rt/blob/master/lib/fuzzer/FuzzerTracePC.cpp#L585

So I assume using ASAN is better than not using it then? WDYT?

**Comment by morehouse** [comment]
Probably.  We have unit tests that definitely benefit from the strcmp hooks, but I never did an A/B test on fuzzer-test-suite.

**Comment by jonathanmetzman** [comment]
> Probably. We have unit tests that definitely benefit from the strcmp hooks, but I never did an A/B test on fuzzer-test-suite.

OK. I'll add one libfuzzer w/asan and one without. If ASAN has significant benefits we can use it for libFuzzer in the future.

**Pull request merged**